Tools/Scripts/webkitdirs.pm

```diff
@@ -412,7 +412,13 @@ sub determineBaseProductDir
             $baseProductDir = $1 if $baseProductDir =~ /SYMROOT\s*=\s*\"(.*?)\";/s;
         }
 
-        undef $baseProductDir unless $baseProductDir =~ /^\//;
+        # Expand tilde in pathnames for compatibility
+        # (https://bugs.webkit.org/show_bug.cgi?id=249442).
+        ($baseProductDir) = bsd_glob($baseProductDir, GLOB_TILDE) if $baseProductDir;
+
+        # webkitdirs can be run from arbitrary directories, so any
+        # user-specified build directory must be absolute.
+        undef $baseProductDir unless defined($baseProductDir) && $baseProductDir =~ /^\//;
```

Comment:
Is it appropriate to print warnings from `webkitdirs.pm`? I feel like I'd like to know if my settings are ignored here.